coderd/workspacebuilds_test.go

```diff
@@ -1334,29 +1342,63 @@ func TestWorkspaceBuildTimings(t *testing.T) {
 			ResourceID: resource.ID,
 		})
 
-		// Then: return a response with empty agent script timings
+		// When: fetching timings for the build
 		ctx, cancel := context.WithTimeout(context.Background(), testutil.WaitLong)
 		t.Cleanup(cancel)
 		res, err := client.WorkspaceBuildTimings(ctx, build.ID)
 		require.NoError(t, err)
+
+		// Then: return a response with empty agent script timings
 		require.Empty(t, res.AgentScriptTimings)
 	})
 
 	// Some workspaces might not have agents. It is improbable, but possible.
 	//nolint:paralleltest
 	t.Run("NoAgents", func(t *testing.T) {
-		// When: fetching timings for a build with no agents
+		// Given: a build with no agents
 		build := makeBuild()
 		dbgen.WorkspaceResource(t, db, database.WorkspaceResource{
 			JobID: build.JobID,
 		})
 
-		// Then: return a response with empty agent script timings
-		// trigger build
+		// When: fetching timings for the build
 		ctx, cancel := context.WithTimeout(context.Background(), testutil.WaitLong)
 		t.Cleanup(cancel)
 		res, err := client.WorkspaceBuildTimings(ctx, build.ID)
 		require.NoError(t, err)
+
+		// Then: return a response with empty agent script timings
 		require.Empty(t, res.AgentScriptTimings)
 	})
+
+	//nolint:paralleltest
+	t.Run("AgentConnectionTimings", func(t *testing.T) {
```

Comment:
Please add a specific case for no agents.

Reply:
We already have a test for this here https://github.com/coder/coder/blob/3e1170c1730cea1d3cf01d1c8ca854584b2f0f4d/coderd/workspacebuilds_test.go#L1362 but I will add a check to verify if the agent connection timings are returning correctly 👍